main.tf

```diff
@@ -156,7 +156,7 @@ resource "azurerm_linux_virtual_machine" "vm_linux" {
 
     content {
       public_key = admin_ssh_key.value.public_key
-      username   = admin_ssh_key.value.username
+      username   = admin_ssh_key.value.username == null ? var.admin_username : admin_ssh_key.value.username
```

Comment:
Could we simplify this expression like this?:

```hcl
username   = coalesce(admin_ssh_key.value.username, var.admin_username)
```

Reply:
Yes, again cleaner and simpler.